src/printbox-ext-plot/PrintBox_ext_plot.mli

```diff
@@ -0,0 +1,61 @@
+(* This file is free software. See file "license" for more details. *)
+
+(** {1 Extend {!PrintBox.t} with plots of scatter graphs and line graphs} *)
+
+type plot_spec =
+  | Scatterplot of {
+      points: (float * float) array;
+      content: PrintBox.t;
+    }
+  | Scatterbag of { points: ((float * float) * PrintBox.t) array }
+  | Line_plot of {
+      points: float array;
+      content: PrintBox.t;
+    }
+  | Boundary_map of {
+      callback: float * float -> bool;
+      content_true: PrintBox.t;
+      content_false: PrintBox.t;
+    }
+  | Map of { callback: float * float -> PrintBox.t }
+  | Line_plot_adaptive of {
+      callback: float -> float;
+      cache: (float, float) Hashtbl.t;
+      content: PrintBox.t;
+    }
+[@@deriving sexp_of]
+
+type graph = {
+  specs: plot_spec list;
+      (** Earlier plots in the list take precedence: in case of overlap, their contents
+          are on top. For HTML, we ensure that framed boxes and grids with bars are opaque. *)
+  x_label: string;  (** Horizontal axis label. *)
+  y_label: string;  (** Vertical axis label. *)
+  size: int * int;
+      (** Size of the graphing area in pixels. Scale for characters is configured by
+          {!scale_size_for_text}. *)
+  no_axes: bool;
```

Comment:
maybe this should be `axes: bool`, skip the negation there and add it in the code?